mempool/mempool_test.go

```diff
@@ -532,6 +742,221 @@ func TestSimpleOrphanChain(t *testing.T) {
 	}
 }
 
+// TestTicketPurchaseOrphan ensures that ticket purchases are orphaned when
+// referenced outputs spent are from missing transactions.
+func TestTicketPurchaseOrphan(t *testing.T) {
+	t.Parallel()
+
+	harness, spendableOuts, err := newPoolHarness(&chaincfg.MainNetParams)
+	if err != nil {
+		t.Fatalf("unable to create test pool: %v", err)
+	}
+
+	tc := &testContext{t, harness}
+
+	// Create a regular transaction from the first spendable output
+	// provided by the harness.
+	tx, err := harness.createTx(spendableOuts[0])
+	if err != nil {
+		t.Fatalf("unable to create transaction: %v", err)
+	}
+
+	// Create a ticket purchase transaction spending the outputs of the
+	// prior regular transaction.
+
+	ticket, err := harness.CreateTicketPurchase(tx, 40000)
+	if err != nil {
+		t.Fatalf("unable to create ticket purchase transaction: %v", err)
+	}
+
+	// Ensure the ticket purchase is accepted as an orphan.
+	_, err = harness.txPool.ProcessTransaction(ticket, true,
+		false, true)
+	if err != nil {
+		t.Fatalf("ProcessTransaction: failed to accept valid "+
+			"orphan %v", err)
+	}
+
+	testPoolMembership(tc, ticket, true, false)
+
+	// Add the regular transaction whose outputs are spent by the ticket purchase
+	// and ensure they all get accepted.  Notice the accept orphans flag is also
+	// false here to ensure it has no bearing on whether or not already existing
+	// orphans in the pool are linked.
+	_, err = harness.txPool.ProcessTransaction(tx,
+		false, false, true)
+	if err != nil {
+		t.Fatalf("ProcessTransaction: failed to accept valid "+
```

Comment:
Since the comments are using 4-space tabs, this should too.

```Go
		t.Fatalf("ProcessTransaction: failed to accept valid transaction %v",
			err)
```